File: 04-Proyecto-LP2Soft/ProyectosJAVA/LP2SoftLogisticaModel/src/pe/edu/pucp/lp2soft/ventas/model/LineaOrdenVenta.java
package pe.edu.pucp.lp2soft.ventas.model;
import pe.edu.pucp.lp2soft.almacen.model.Producto;
public class LineaOrdenVenta {
    private int idLineaOrdenVenta;
    private Producto producto;
    private int cantidad;
    private double subtotal;
    private boolean activo;

    public LineaOrdenVenta(Producto producto, int cantidad) {
        this.producto = producto;
        this.cantidad = cantidad;
    }
    
    public int getIdLineaOrdenVenta() {
        return idLineaOrdenVenta;
    }

    public void setIdLineaOrdenVenta(int idLineaOrdenVenta) {
        this.idLineaOrdenVenta = idLineaOrdenVenta;
    }

    public Producto getProducto() {
        return producto;
    }

    public void setProducto(Producto producto) {
        this.producto = producto;
    }

    public int getCantidad() {
        return cantidad;
    }

    public void setCantidad(int cantidad) {
        this.cantidad = cantidad;
    }

    public double getSubtotal() {
        return subtotal;
    }

    public void setSubtotal(double subtotal) {
        this.subtotal = subtotal;
    }

    public boolean isActivo() {
        return activo;
    }

    public void setActivo(boolean activo) {
        this.activo = activo;
    }
    
    public void calcularSubtotal(){
        this.subtotal = cantidad * producto.getPrecio();
    }
}

File: 04-Codigo-Clases/ProyectosNET/LP2Soft/LP2SoftLogisticaModel/Ventas/LineaOrdenVenta.cs
using LP2SoftLogisticaModel.Almacen;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LP2SoftLogisticaModel.Ventas
{
    public class LineaOrdenVenta
    {
        private int _idLineaOrdenVenta;
        private Producto _producto;
        private int _cantidad;
        private double _subtotal;
        private bool _activo;

        public int IdLineaOrdenVenta { get => _idLineaOrdenVenta; set => _idLineaOrdenVenta = value; }
        public Producto Producto { get => _producto; set => _producto = value; }
        public int Cantidad { get => _cantidad; set => _cantidad = value; }
        public double Subtotal { get => _subtotal; set => _subtotal = value; }
        public bool Activo { get => _activo; set => _activo = value; }
    }
}
